manifests/server.pp

```diff
@@ -384,8 +388,10 @@
   Array[Stdlib::Absolutepath, 1] $envs_dir = $puppet::server_envs_dir,
   Optional[Stdlib::Absolutepath] $envs_target = $puppet::server_envs_target,
   Variant[Undef, String[0], Array[Stdlib::Absolutepath]] $common_modules_path = $puppet::server_common_modules_path,
-  Pattern[/^[0-9]{3,4}$/] $git_repo_mode = $puppet::server_git_repo_mode,
+  Pattern[/^[0-9]{3,4}$/] $git_repo_hook_mode = $puppet::server_git_repo_hook_mode,
   Stdlib::Absolutepath $git_repo_path = $puppet::server_git_repo_path,
+  Optional[String] $git_repo_source = $puppet::server_git_repo_source,
```

Comment:
This is actually a pretty massive difference in functionality. The old mode created a bare repository which you were intended to push to. Do you have have any idea how the hook interacts with that?

Before this you had:
* Empty repo is initialized
* You push all data
* Hook creates all environments

If you now provide something to clone from, do you end up with all environments? If not, you can't trigger it unless you create some dummy commit, right?

Reply:
As you say: without the parameter the behavior is unchanged. Often I end up then writing one patch to move over and then another to change the behavior. So we could leave out the source parameter at first and then introduce that as a new parameter in a separate feature.

Reply:
I'll split this into two as my test VMs seem to be sad today.

Reply:
Opened https://projects.theforeman.org/issues/35945 to track the `git clone` bits.